### PR TITLE
Add new entry on virtual machines

### LINKED ISSUE #710: [Entry] General: Virtual machines
### Reason/inspiration (optional)

_No response_

### Entry Type (select all that apply)

New Topic Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I added the virtual-machines concept and virtual-machines.md file inside. The file (1) introduces VMs, (2) explains how VMs work, (3) lists the benefits of using VMs, and (4) notes who uses VMs.

Closes #710 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:18]:
```suggestion
A single server or physical computer can run multiple VMs at the same time. Each VM has its own operating system and functions as an independent compute resource, even though they may share the same host server. For example, a computer running macOS could use a VM to run a separate instance of the Windows operating system from the same computer.
```

For Docs entries, let's try to avoid a "you" voice or "we" voice whenever possible, in favor of a more declarative, manual tone.

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:3]:
```suggestion
Description: 'Virtual machines are resources that can perform the same tasks as physical computers.'
```

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:10]: Let's include any tags that come from the Docs [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) file. What I'd suggest is the following:

```suggestion
  - 'Servers'
  - 'Cybersecurity'
```

Though I want to check with my colleagues and see if we can at least add a `'Virtualization'` tag 😄

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:30]:
```suggestion
### Improved Hardware Utilization
```

For Docs entries, even the subheaders use "Title Case". Can we apply this edit to the remainder of the file?

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [reply content/general/concepts/virtual-machines/virtual-machines.md:20]: It's recommended we stick with the abbreviated term after the first mention.

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:22]:
```suggestion
VMs run on top of the operating system of the host machine, similar to other applications on a physical computer. All VMs on a single host share the same physical resources, such as memory and storage.
```

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:24]:
```suggestion
Multiple VMs can be managed with a software known as a “hypervisor”. A hypervisor is used to create, manage, and run the VMs on a host machine. The hypervisor is used to manage hardware resources from the host machine, which enables these resources to be shared amongst multiple VMs.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@YanisaHS Finished with my review and left some comments/suggestions 😄 Thanks for making this new entry, btw!

**Comment by YanisaHS** [comment]
@Dusch4593 Thanks for your detailed feedback!! Done 🥳

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:40]:
```suggestion
VMs can also improve security because VMs are isolated from the host computer and each other. If an attacker gained access to a single VM, they won’t have access to the host machine or other VMs that share the same host.
```

**Review by Dusch4593 — APPROVED** [review]
@YanisaHS LGTM! (just one last trailing suggestion 😄)

**Comment by YanisaHS** [comment]
@Dusch4593 Done! 😄

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (SSwiniarski) [inline content/general/concepts/virtual-machines/virtual-machines.md:3]:
```suggestion
Description: 'Virtual machines are software resources that can perform the same tasks as physical computers.'
```

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (SSwiniarski) [inline content/general/concepts/virtual-machines/virtual-machines.md:14]:
```suggestion
A **virtual machine (VM)** is a software resource that can perform the same computing functions as a physical computer system, such as running applications or storing data. However, VMs do not use physical hardware to perform these tasks. Instead, VMs exist as code and use software-defined versions of hardware that are installed on a host [server](https://www.codecademy.com/resources/docs/general/server).
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@YanisaHS, I did a second review and only had two suggestions.

**Comment by YanisaHS** [comment]
@SSwiniarski Thanks for the review! Just committed those changes 🥳

**Review by yangc95 — APPROVED** [review]
LGTM

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:38]:
```suggestion
Using separate VMs is considered a secure way to explore potentially malicious applications or viruses. Because they are separate environments, VMs can be used for these tasks without the risk of compromising the host machine.
```

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:40]:
```suggestion
This isolation improves security. If an attacker gained access to a single VM, they wouldn't have access to the host machine or other VMs that share the same host.
```

File: content/general/concepts/virtual-machines/virtual-machines.md
Comment (Dusch4593) [inline content/general/concepts/virtual-machines/virtual-machines.md:32]:
```suggestion
### Organized Development and Test Environments
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@YanisaHS Just a few last super small comments/suggestions from me 😄

**Comment by YanisaHS** [comment]
@Dusch4593 Done! 🥳

**Review by Dusch4593 — APPROVED** [review]
@YanisaHS LGTM for merge! 🚀

I made a few small edits in c3099d8df430e3c45da08b900458938346be3158:

- Added `'Cloud Computing'` and `'Operating Systems'` tags.
- Revised link to Cloud Computing (now its own topic; no longer under General).

**Comment by Dusch4593** [comment]
@SSwiniarski Anything left to add, change, or check before we merge?

**Review by SSwiniarski — APPROVED** [review]
Looks good to me.

**Comment by YanisaHS** [comment]
@Dusch4593 Would it be possible to add the hacktoberfest tag to this PR? Not sure how it works, but trying to do that thing 🥳 I think this one isn’t showing up in my account because it doesn’t have the tag.

**Comment by Dusch4593** [comment]
> @Dusch4593 Would it be possible to add the hacktoberfest tag to this PR? Not sure how it works, but trying to do that thing 🥳 I think this one isn’t showing up in my account because it doesn’t have the tag.

I can certainly add the tag, though I'm not 100% certain it would be counted because the PR was created prior to Oct. 1st. ([source](https://hacktoberfest.com/participation/#pr-mr-details:~:text=Your%20PR/MRs%20must%20be%20created%20between%20October%201%20and%20October%2031%20(in%20any%20time%20zone%2C%20UTC%2D12%20thru%20UTC%2B14).))

**Pull request merged**
